File: Labs/Lab09/src/main/java/com/efekurucay/lab09/model/Course.java
package com.efekurucay.lab09.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnore;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Course {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private String code;

    @ManyToOne 
    @JoinColumn(name = "teacher_id")
    @JsonIgnore
    private Teacher teacher;

    @ManyToMany(mappedBy = "courses")
    @JsonIgnore
    private List<Student> students;
}

// course id , name, kod, teacher, ogrenci

File: Labs/Lab09 - 10 std- teacher- course/frontend/src/app/models/course.ts
import { CourseStatus } from './course-status';
import { Student } from './student';
import { Teacher } from './teacher';

export interface Course {
  id?: number;
  name: string;
  code: string;
  description: string;
  status: CourseStatus;
  teacher?: Teacher;
  students?: Student[];
} 